bpf/lib/nat.h

```diff
@@ -1064,10 +1061,85 @@ snat_v4_nat(struct __ctx_buff *ctx, const struct ipv4_nat_target *target, __s8 *
 	if (ret < 0)
 		return ret;
 
-rewrite_egress:
 	return snat_v4_rewrite_egress(ctx, &tuple, state, off, ipv4_has_l4_header(ip4));
 }
 
+static __always_inline __maybe_unused int
+snat_v4_rev_nat_handle_icmp_frag_needed(struct __ctx_buff *ctx, __u64 off)
+{
+	struct ipv4_ct_tuple tuple = {};
+	struct ipv4_nat_entry *state;
+	struct iphdr iphdr;
```

Comment:
Just for context - when initially adding this code (a short while back), there was concern whether we should share the CT tuple (for processing the inner & outer packet), or have it cleanly separated. And how it would impact complexity.